### PR TITLE
update tonic tutorial to use tonic version 1.0.2

### PR DESCRIPTION
**Pull request by biphasic**
I updated the previous notebook. I opted for showing that caching is useful by timing some of the loading times. Feel free to shorten that part if too long!

### CODE REVIEW

File: examples/legacy/tutorial_5_neuromorphic_datasets.ipynb [hunk 1]
```diff
@@ -60,89 +86,323 @@
...
+   "execution_count": 7,
+   "id": "f47e0798-5259-491a-9d3a-59b13b1b0983",
    "metadata": {},
    "outputs": [],
    "source": [
     "from torch.utils.data import DataLoader\n",
-    "trainloader = DataLoader(trainset, shuffle=True)\n",
+    "from tonic import CachedDataset\n",
+    "\n",
+    "cached_trainset = CachedDataset(trainset, cache_path='./cache/nmnist/train')\n",
```
Comment (jeshraghian) [inline examples/legacy/tutorial_5_neuromorphic_datasets.ipynb:203]: When running in colab, the `cache_path` is created, but the hdf5 file is not. I get the following error (same issue in the `fast_dataloading` tutorial in the Tonic docs):

> OSError: Unable to open file (unable to open file: name = 'data/cache/nmnist/train/0_0.hdf5', errno = 2, error message = 'No such file or directory', flags = 0, o_flags = 0)

**Review by jeshraghian — COMMENTED** [review]
This is looking great so far man - thanks so much! 
Caching only works for me. 

I ran into an issue when calling `CachedDataset()`. 
It appears as though the hdf5 file isn't created when running in Colab. Flagged the specific line in the review.

File: examples/legacy/tutorial_5_neuromorphic_datasets.ipynb (hunk 1 above)
Comment (biphasic) [reply examples/legacy/tutorial_5_neuromorphic_datasets.ipynb:186]: fixed it in version 1.0.4
the issue was an older version of h5py

**Comment by jeshraghian** [comment]
Works perfectly. I'll update the tutorial + send it through to you before publishing.
Thanks!

**Pull request merged**